internal/controller/utils.go

```diff
@@ -51,3 +51,17 @@ func HasChanged(action ssa.Action) bool {
 		return true
 	}
 }
+
+// TrimError returns the error message trimmed to a maximum length of 15K characters
+// which should fit in the status condition message.
+func TrimError(err error) string {
+	if err == nil {
+		return ""
+	}
+	errMsg := err.Error()
+	maxLength := 15000
```

Comment:
```suggestion
	const maxLength = 15000
```

What made you pick `15000` instead of `32765`?

Reply:
Some errors end up twice in the status and it could reach the max size of the whole object.